homeassistant/components/tado/coordinator.py

```diff
@@ -0,0 +1,390 @@
+"""Coordinator for the Tado integration."""
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+import logging
+from typing import Any
+
+import PyTado
+import PyTado.exceptions
+from PyTado.interface import Tado
+from requests import RequestException
+
+from homeassistant.components.climate import PRESET_AWAY, PRESET_HOME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, HomeAssistantError
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, INSIDE_TEMPERATURE_MEASUREMENT, PRESET_AUTO, TEMP_OFFSET
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=4)
+SCAN_INTERVAL = timedelta(minutes=5)
+SCAN_MOBILE_DEVICE_INTERVAL = timedelta(seconds=30)
+
+
+class TadoDataUpdateCoordinator(DataUpdateCoordinator):
```

Comment:
```suggestion
class TadoDataUpdateCoordinator(DataUpdateCoordinator[..]):
```
Let's add the type of `coordinator.data` to the typeslot of the coordinator